File: app/src/main/java/org/worshipsongs/fragment/SettingsPreferenceFragment.java
package org.worshipsongs.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;

import org.worshipsongs.R;
import org.worshipsongs.WorshipSongApplication;
import org.worshipsongs.activity.UserSettingActivity;
import org.worshipsongs.listener.ThemePreferenceListener;
import org.worshipsongs.preference.LanguagePreference;
import org.worshipsongs.preference.PreferenceListener;
import org.worshipsongs.preference.ThemeListPreference;

import static org.worshipsongs.CommonConstants.THEME_KEY;
/**
 * Author:Seenivasan, Madasamy
 * version:1.0.0
 */

public class SettingsPreferenceFragment extends PreferenceFragment implements PreferenceListener
{
    private UserSettingActivity userSettingActivity = new UserSettingActivity();

    @Override
    public void onCreate(final Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.settings);
        languagePreference();
        themePreference();
        resetPreferenceSettings("resetDialog");
    }

    private void languagePreference()
    {
        LanguagePreference resetDialogPreference = (LanguagePreference) findPreference(
                "languagePreference");
        resetDialogPreference.setPreferenceListener(this);
    }

    private void themePreference()
    {
        ThemeListPreference themeListPreference = (ThemeListPreference) findPreference(THEME_KEY);
        themeListPreference.setOnPreferenceChangeListener(new ThemePreferenceListener(userSettingActivity,
                this));
    }

    public void resetPreferenceSettings(String preferenceKey)
    {
        Preference resetDialogPreference = findPreference(preferenceKey);
        resetDialogPreference.setOnPreferenceChangeListener(
                new Preference.OnPreferenceChangeListener()
                {
                    @Override
                    public boolean onPreferenceChange(Preference preference, Object newValue)
                    {
                        Intent startIntent = new Intent(WorshipSongApplication.getContext(),
                                UserSettingActivity.class);
                        startIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                        userSettingActivity.activityFinish();
                        startActivity(startIntent);
                        return false;
                    }
                });
    }

    public void setUserSettingActivity(UserSettingActivity userSettingActivity)
    {
        this.userSettingActivity = userSettingActivity;
    }

    @Override
    public void onSelect()
    {
        userSettingActivity.invalidateOptionsMenu();
        userSettingActivity.finish();
        Intent startIntent = new Intent(WorshipSongApplication.getContext(),
                UserSettingActivity.class);
        startActivity(startIntent);

    }
}

File: app/src/main/java/org/worshipsongs/fragment/SettingsPreferenceFragment.kt
package org.worshipsongs.fragment


import android.content.Intent
import android.os.Bundle
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import org.worshipsongs.CommonConstants.THEME_KEY
import org.worshipsongs.R
import org.worshipsongs.WorshipSongApplication
import org.worshipsongs.activity.UserSettingActivity
import org.worshipsongs.listener.ThemePreferenceListener
import org.worshipsongs.preference.LanguagePreference
import org.worshipsongs.preference.PreferenceListener
import org.worshipsongs.preference.ThemeListPreference
import org.worshipsongs.service.ResetDefaultSettingsService

/**
 * @author Madasamy
 * @since 1.0.0
 */

class SettingsPreferenceFragment : PreferenceFragmentCompat(), PreferenceListener
{

    private var userSettingActivity = UserSettingActivity()

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?)
    {
        setPreferencesFromResource(R.xml.settings, rootKey)
        languagePreference()
        themePreference()
        resetPreferenceSettings()
    }

    private fun languagePreference()
    {
        val preference = findPreference<LanguagePreference>("languagePreference")
        if (preference is LanguagePreference)
        {
            preference.setPreferenceListener(this)
        }
    }


    private fun themePreference()
    {
        val themeListPreference = findPreference<ThemeListPreference>(THEME_KEY)
        themeListPreference!!.onPreferenceChangeListener = ThemePreferenceListener(userSettingActivity, this)
    }

   private fun resetPreferenceSettings()
    {
        val resetDialogPreference = findPreference<ResetDefaultSettingsService>("resetDialog")
        resetDialogPreference!!.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, newValue ->
            val startIntent = Intent(WorshipSongApplication.context, UserSettingActivity::class.java)
            startIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            userSettingActivity.activityFinish()
            startActivity(startIntent)
            false
        }
    }

    fun setUserSettingActivity(userSettingActivity: UserSettingActivity)
    {
        this.userSettingActivity = userSettingActivity
    }

    override fun onSelect()
    {
        userSettingActivity.invalidateOptionsMenu()
        userSettingActivity.finish()
        val startIntent = Intent(WorshipSongApplication.context, UserSettingActivity::class.java)
        startActivity(startIntent)
    }
}